tests/Integration/Behaviour/Features/Context/Domain/CartFeatureContext.php

```diff
@@ -727,6 +757,53 @@ public function assertReductionValueOfCartCartRule(
         throw new RuntimeException(sprintf('Voucher was %s not found in cart', $voucherCode));
     }
 
+    /**
+     * @Then cart :cartReference should contain :quantity products
+     *
+     * @param string $cartReference
+     * @param int $quantity
+     */
+    public function assertCartNumberOfProducts(string $cartReference, int $quantity)
+    {
+        $cartInfo = $this->getCartInformationByReference($cartReference);
+
+        $cartProductsQuantity = \count($cartInfo->getProducts());
```

Comment:
I'm surprised `count` needs to be backslashed

Reply:
ow no need. I used to do this for PHP native functions. I will remember the explanation. It's about preprocessing

Reply:
I don't think it is needed